File: lab-5/src/com/company/commands/Command.java
package com.company.commands;

import com.company.collection.MusicBandStorage;
import com.company.container.Message;
import com.company.util.IOStream;

public abstract class Command {
    protected final MusicBandStorage target;
    protected final String cmdName;
    protected final String description;
    protected final IOStream ioStream;

    public Command(MusicBandStorage target, String cmdName, IOStream ioStream, String description) {
        this.target = target;
        this.cmdName = cmdName;
        this.ioStream = ioStream;
        this.description = description;
    }

    public String getCmdName() {
        return cmdName;
    }

    public String getDescription() {
        return description;
    }

    public abstract Message exec(Message message);

    // Message help(); -> ??? description


}

File: lab-5-kotlin/src/main/kotlin/com/github/e1turin/lab5/commands/Command.kt
package com.github.e1turin.lab5.commands

import com.github.e1turin.lab5.collection.MusicBandStorage
import com.github.e1turin.lab5.containers.Message
import com.github.e1turin.lab5.util.IOStream

abstract class Command(
    //    protected val target: MusicBandStorage,
    val cmdName: String,
    //    protected var ioStream: IOStream,
    val description: String
) {

    abstract fun exec(arg: String, target: MusicBandStorage, ioStream: IOStream): Message// Message help(); -> ??? description

}
